src/controller/securemgr/authorizer/authorizer.go

```diff
@@ -0,0 +1,141 @@
+/*******************************************************************************
+* Copyright 2020 Samsung Electronics All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*******************************************************************************/
+package authorizer
+
+import (
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/casbin/casbin"
+)
+
+// AuthorizerImpl structure
+type AuthorizerImpl struct{}
+
+const (
+	rbacPolicyFileName = "policy.csv"
+	policy_template    = "p, admin, /*, *\n" +
+		"p, member, /api/v1/orchestration/services, *\n"
+	rbacAuthModelFileName = "auth_model.conf"
+	auth_model_template   = "[request_definition]\n" +
+		"r = sub, obj, act\n\n" +
+		"[policy_definition]\n" +
+		"p = sub, obj, act\n\n" +
+		"[policy_effect]\n" +
+		"e = some(where (p.eft == allow))\n\n" +
+		"[matchers]\n" +
+		"m = r.sub == p.sub && keyMatch(r.obj, p.obj) && (r.act == p.act || p.act == \"*\")\n"
+)
+
+var (
+	logPrefix             = "[securemgr: RBAC]"
+	rbacPolicyFilePath    = ""
+	rbacAuthModelFilePath = ""
+	authorizerIns         *AuthorizerImpl
+	initialized           = false
+	users                 Users
+	enf                   *casbin.Enforcer
+)
+
+func init() {
+	authorizerIns = new(AuthorizerImpl)
+}
+
+type User struct {
+	Name string
+	Role string
+}
+
+// Users is a list of users
+type Users []User
+
+// findByName returns the user with the given name
+func (u Users) findByName(name string) (User, error) {
+	for _, user := range u {
+		if user.Name == name {
+			return user, nil
+		}
+	}
+	return User{}, errors.New("User is not found")
+}
+
+// Init sets the environments for securemgr
+func Init(rbacRulePath string) {
+	if _, err := os.Stat(rbacRulePath); err != nil {
```

Comment:
Good questions from a developer. 

1. When you develop on a device it is possible to do it with `build.sh` script. But imagine that you just downloaded the edge-orchestration.tar file, in this case you need to create the configuration manually. 

2. Regarding prerequisites: I think this is bad, the user needs to do a lot of additional operations, so I decided that the system should not rely on the correctness of the configuration (the user may be mistaken).
P.S. From the security side, we will also need to ask the user to manually configure the file access control system. And in general, it will be very difficult to deploy the edge-orchestration.

3. But of course we can do as you suggest, if my arguments are not so important.
Please write what you think about it! Thank you!

Reply:
> Why don't we make the rbacRulePath folder through the build script like below rather than checking it on Init period?
> https://github.com/lf-edge/edge-home-orchestration-go/blob/59c7f490604e4fceb6df59325ac834452654fd74/build.sh#L413


> Good questions from a developer.
> 
> 1. When you develop on a device it is possible to do it with `build.sh` script. But imagine that you just downloaded the edge-orchestration.tar file, in this case you need to create the configuration manually.
> 2. Regarding prerequisites: I think this is bad, the user needs to do a lot of additional operations, so I decided that the system should not rely on the correctness of the configuration (the user may be mistaken).
>    P.S. From the security side, we will also need to ask the user to manually configure the file access control system. And in general, it will be very difficult to deploy the edge-orchestration.
> 3. But of course we can do as you suggest, if my arguments are not so important.
>    Please write what you think about it! Thank you!

@t25kim It seems to me that the current PR can go forward. Please let me know if we can accept this PR w.r.t. your previous suggestion. ^^

Reply:
@tdrozdovsky's idea is more convenient for developers. I agree his idea!

Reply:
> @tdrozdovsky's idea is more convenient for developers. I agree his idea!

@t25kim Thank you for the prompt and positive feedback! Will accept the PR then! @tdrozdovsky Thank you for your consistent proposal for our project especially in a security domain.